src/python/tests/core/test_scheduler.py

```diff
@@ -9,25 +9,119 @@
 import os
 import time
 import uuid
-
+import aiopg
 import pytest
+from foglamp.core.scheduler.scheduler import Scheduler
+from foglamp.core.scheduler.entities import IntervalSchedule, Task, Schedule, TimedSchedule, ManualSchedule, StartUpSchedule
+from foglamp.core.scheduler.exceptions import ScheduleNotFoundError
+from foglamp.core.server import Server
+from foglamp.storage.storage import Storage
 
-from foglamp.core.scheduler import (Scheduler, IntervalSchedule, ScheduleNotFoundError, Task,
-                                    Schedule, TimedSchedule, ManualSchedule, StartUpSchedule, Where)
-
-
-__author__ = "Terris Linenbach"
+__author__ = "Terris Linenbach, Amarendra K Sinha"
 __copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
 __license__ = "Apache 2.0"
 __version__ = "${VERSION}"
 
+_CONNECTION_STRING = "dbname='foglamp' user='foglamp'"
+_FOGLAMP_ROOT = os.getenv("FOGLAMP_ROOT", default='/home/foglamp/foglamp/FogLAMP')
+_STORAGE_DIR = os.path.expanduser(_FOGLAMP_ROOT + '/services/storage')
+
+"""
+    _is_management_started, _address, _host, _m_port, _app, _server_handler, _server are module level variables.
+
+    start_storage(), stop_storage() and start_management() are module level functions.
+
+    setup_module() and teardown_module() are module level setup and teardown methods but we are using teardown_module()
+    only.
+
+    setup has been moved into each individual test as start_management(), essential part of setup, is a coro and is
+    called by the event loop and setup_module() is never called by the event loop. Also, though setup has been defined
+    in each test, as we do not know the order of the execution of the tests, BUT is executed only once.
+"""
+
+_is_management_started = False
+_address = None
+_host = '0.0.0.0'
+_m_port = 0
+_app = None
+_server_handler = None
+_server = None
+
+def start_storage(host, m_port):
+    try:
+        cmd_with_args = ['./storage', '--address={}'.format(host),
+                         '--port={}'.format(m_port)]
+        import subprocess
+        subprocess.call(cmd_with_args, cwd=_STORAGE_DIR)
+    except Exception as ex:
+        pass
+
+def stop_storage():
+    try:
+        Storage().shutdown()
```

Comment:
make a http client call. if you want to keep this test infra. Need to review this test structure